schema.go

```diff
@@ -335,31 +335,34 @@ type RecordSchema struct {
 
 	isError bool
 	fields  []*Field
+	doc     string
 }
 
 // NewRecordSchema creates a new record schema instance.
-func NewRecordSchema(name, space string, fields []*Field) (*RecordSchema, error) {
+func NewRecordSchema(name, space string, doc string, fields []*Field) (*RecordSchema, error) {
```

Comment:
You cannot add the doc in the constructor, as this would make this a backward breaking change. I would suggest adding a second function `AddDoc` instead.